packages/table-core/src/features/RowSelection.ts

```diff
@@ -436,7 +436,7 @@ export const RowSelection: TableFeature = {
       ).length
       return (
         totalSelected > 0 &&
-        totalSelected < table.getFilteredRowModel().flatRows.length
+        totalSelected <= table.getFilteredRowModel().flatRows.length
```

Comment:
This is not the correct fix. We should just get rid of all the logic that returns false if all rows are selected.